src/commonMain/kotlin/com/algolia/search/model/settings/Settings.kt

```diff
@@ -357,11 +357,18 @@ public data class Settings(
     /**
      *  Lets you store custom data in your indices.
      */
-    @SerialName(KeyUserData) var userData: JsonObject? = null
+    @SerialName(KeyUserData) var userData: JsonObject? = null,
+    /**
+     * This parameter configures the segmentation of Japanese text at indexing time.
+     * Accepted value: [Language.Japanese]
+     * Input data to index is treated as Japanese text.
+     */
+    @SerialName(KeyIndexLanguages) var indexLanguages: List<Language>? = null
```

Comment:
Do we want to limit it to supported language? If yes, we should create `IndexLanguages` as a subset of `Languages`

Reply:
Up to you guys but looking a bit in the future, it seems like more maintenance on your side for nothing (IIRC the error is pretty self-explanatory when the engine sees a language it doesn't support yet).

Reply:
Yes I thought about it, but deemed the cost greater than the benefit.